### ISSUE TITLE
Cohort Criteria Range Check

### ISSUE THREAD

**Issue by anthonysena** [OHDSI/Atlas#532]
### Expected behavior
Entering a criteria that cannot be satisfied logically should raise a warning in the interface to alert the user. For example, consider the following example criteria:

At least 1 occurrences of a condition occurrence of <condition> starting between 28 days Before and 365 days Before event index date

![image](https://user-images.githubusercontent.com/12902366/35240289-fb30f2a4-ff80-11e7-9ce8-f62cbf1a63bb.png)

This results in the following SQL logic:

`A.START_DATE >= DATEADD(day,-28,P.START_DATE) and A.START_DATE <= DATEADD(day,-365,P.START_DATE)`

And this logically cannot be satisfied.

### Actual behavior

The interface accepts the input and the criteria fails to find anyone (as it should). 

### Steps to reproduce behavior

Details are in the expected behavior

**Comment by anthonysena** [comment]
Linking to #1933 as this may exist now and if not, it would be a useful addition.

**Comment by pbr6cornell** [comment]
I agree this is one of the warnings that should appear in the Warnings tab.  @pavgra , where can I find the documentation of existing warnings to confirm that its already covered?

**Comment by anthonysena** [comment]
From discussion with @olga-ganina - we'll review this issue to make sure that the original criteria check is covered under the circe library of checks here: https://github.com/OHDSI/circe-be/tree/master/src/main/java/org/ohdsi/circe/check/checkers. We'll also review how to best document and test those checks.
